src/library_glfw.js

```diff
@@ -671,6 +672,64 @@ var LibraryGLFW = {
       win.scrollFunc = cbfun;
     },
 
+    setDropCallback: function(winid, cbfun) {
+      var win = GLFW.WindowFromId(winid);
+      if (!win) return;
+      win.dropFunc = cbfun;
+    },
+
+    onDrop: function(event) {
+      if (!GLFW.active.dropFunc) return;
+
+      event.preventDefault();
+
+      var filenames = allocate(new Array(event.dataTransfer.files.length*4), 'i8*', ALLOC_NORMAL);
+      var filenamesArray = [];
+      var count = event.dataTransfer.files.length;
+
+      // Read and save the files to emscripten's FS
+      // TODO: lazily load?
+      var written = 0;
+      for (var i = 0; i < count; ++i) {
+        (function(file) {
+          var reader = new FileReader();
+          reader.onload = function(e) {
+            var data = e.target.result;
+            var path = file.name; // TODO: to a new directory?
```

Comment:
In my testing (latest Chrome, Firefox, and Safari all on macOS), `file.name` is the base name (e.g. `file.txt`). Changed to save in `/.glfw_dropped_files/` and replace slashes as suggested.

Overwriting existing files of the same name is another open question. Should the library take care to avoid overwriting a previously-dropped file with an identical name? Maybe, maybe not. Could check if the file exists, if so, change the filename/path... or choose a random filename/directory, but we want to preserve the filename as much as possible since it may contain vital information. Overwriting files with the same name seems not too unreasonable to me, a common use case could be repeatedly dragging and dropping the same file, after editing it on the user's system, so overwriting would be useful/expected (assuming the ported app did not add the `#ifdef __EMSCRIPTEN__` ... `unlink()` call  yet, after handling the dropped file). I'd lean towards simplicity here, perhaps choosing a randomly-named subdirectory of /.glfw_dropped_files/ would be sufficient.